mllib/src/main/scala/org/apache/spark/ml/classification/NaiveBayes.scala

```diff
@@ -280,15 +381,35 @@ class NaiveBayesModel private[ml] (
    * This precomputes log(1.0 - exp(theta)) and its sum which are used for the linear algebra
    * application of this condition (in predict function).
    */
-  private lazy val (thetaMinusNegTheta, negThetaSum) = $(modelType) match {
-    case Multinomial => (None, None)
+  @transient private lazy val (thetaMinusNegTheta, negThetaSum) = $(modelType) match {
     case Bernoulli =>
       val negTheta = theta.map(value => math.log1p(-math.exp(value)))
       val ones = new DenseVector(Array.fill(theta.numCols) {1.0})
       val thetaMinusNegTheta = theta.map { value =>
         value - math.log1p(-math.exp(value))
       }
-      (Option(thetaMinusNegTheta), Option(negTheta.multiply(ones)))
+      (thetaMinusNegTheta, negTheta.multiply(ones))
+    case Multinomial => (null, null)
+    case Gaussian => (null, null)
```

Comment:
You could collapse these two cases. Or should this be an error? would these ever be referenced outside the Bernoulli case?

Reply:
I think causing an error is ok, avaliable `thetaMinusNegTheta, negThetaSum` should only be referenced in Bernoulli case.